providers/CENNZApiProvider.tsx

```diff
@@ -0,0 +1,28 @@
+import { Api } from "@cennznet/api";
+import { createContext, useState, PropsWithChildren, useContext } from "react";
+
+type CENNZApiContextType = {
+  api: Api;
+  updateApi: Function;
+};
+
+const CENNZApiContext = createContext<CENNZApiContextType>(null);
+
+export default function CENNZApiProvider({ children }: PropsWithChildren<{}>) {
+  const [api, setApi] = useState<Api>(null);
```

Comment:
actually man it would be great to expose the RX API here as well if possible. That way we can do subscriptions for stream data when certain components require it. I think it's gonna solve a lot of the global redux stuff that's currently setup in cennzx

Reply:
@eupston can you give us an example of the RX Api? how's it used atm?

Reply:
yeah so API RX is basically just Observable under the hood. Atm cennzx uses it for almost everything related to getting data off the blockchain. Many of these things like generic assets types probably don't need it imo, but certain elements would like getting the current amount of tokens in the liquidity pool, because it's high volatility data.

Reply:
@eupston got it, so it's an entire different class called `ApiRx`.

```
const {ApiRx} = require('@cennznet/api')
api = await ApiRx.create({
    provider: 'wss://nikau.centrality.me/public/ws'
}).toPromise();
```

@aidan-starke I think the simplest way is to do another `useState` in the same `CENNZApiProvider`, what do you reckon?

Reply:
yup, sounds good I'll add that now

Reply:
nice yeah! thanks guys

Reply:
done, at the moment apiRx is of type Observable<ApiRx>, is it better to use the .toPromise() function and have apiRx as type Promise<ApiRx> ?

Reply:
yeah that'd be great actually, check out this little example:
https://github.com/cennznet/typescript-react-app-cennznet-dApp-template/blob/master/src/App.tsx